src/main/java/com/algorand/algosdk/util/abi/types/TypeUfixed.java

```diff
@@ -0,0 +1,44 @@
+package com.algorand.algosdk.util.abi.types;
+
+import java.util.Objects;
+
+public class TypeUfixed extends Type {
+    public final int bitSize;
+    public final int precision;
+
+    public TypeUfixed(int size, int precision) {
+        if (size < 8 || size > 512 || size % 8 != 0)
+            throw new IllegalArgumentException(
+                    "ufixed initialize failure: bitSize should be in [8, 512] and bitSize moe 8 = 0"
```

Comment:
nit: `mod` instead of `moe`, also consider using `equals` or `==` instead of `=`